debezium-core/src/main/java/io/debezium/document/ComparableValue.java

```diff
@@ -203,6 +203,7 @@ public Float asFloat() {
             double raw = ((Double) value).doubleValue();
             if (isValidFloat(raw)) return Float.valueOf((float) raw);
         }
+        if (value instanceof Number) return ((Number)value).floatValue();
```

Comment:
`floatValue()` seems prone to precision loss, shouldn't we call `doubleValue()` instead?

Reply:
Yes, but the result of the call should be `float` so it is probably unnecessary to call `doubleValue()` and then in next step to convert it to `float`.

Reply:
Ah yes. I missed that we're in `asFloat()` already :)